### PR TITLE
add bionic support, fix race condition in charm start

### PR DESCRIPTION
**Pull request by axinojolais**
see commit messages for more details

### CODE REVIEW

File: reactive/filebeat.py [hunk 1]
```diff
@@ -31,17 +32,25 @@ def install_filebeat():
...
     # Ensure ssl files match config each time we render a new template
     manage_filebeat_logstash_ssl()
     remove_state('beat.render')
+
     if connections:
+        if cfg_original_hash != cfg_new_hash:
+            service('restart', 'filebeat')
         status_set('active', 'Filebeat ready.')
+    else:
+        service('stop', 'filebeat')
```
Comment (kwmonroe) [inline reactive/filebeat.py]: Please set the charm status in this `else` clause with something like:
```
status_set('waiting', 'Waiting for connections.')
```

**Comment by axinojolais** [comment]
status_set added. When I tested the change, after removing the elasticsearch relation, the status was set back to "waiting" with message "Waiting for: elasticsearch, logstash or kafka." from the beats_base layer anyway.

**Comment by kwmonroe** [comment]
> the status was set back to "waiting" with message "Waiting for: elasticsearch, logstash or kafka." from the beats_base layer anyway.

@axinojolais you're right!  sorry about that, i didn't think about the beats-base layer controlling status here.  In any case, it's nice that the status will be set immediately when filebeat shuts down -- maybe it'll prevent a stale status for the short time between shutdown and the beats-base handler running.

**Comment by kwmonroe** [comment]
This has been released to the edge channel here:

https://jujucharms.com/u/filebeat-charmers/filebeat/4

Please note that we're changing the charm owner from `containers` to `filebeat-charmers`.  Until the transition is complete, you'll need to use the full namespace to deploy this charm, i.e.:

```
juju deploy ~filebeat-charmers/filebeat-4
```

**Pull request merged**